### PR TITLE
Support VPN profiles

### PR DESCRIPTION
**Pull request by annakhm**
Signed-off-by: Anna Khmelnitsky <akhmelnitsky@vmware.com>

### CODE REVIEW

File: nsxt/resource_nsxt_policy_ipsec_vpn_dpd_profile.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+	_, err := client.Get(id)
+	if err == nil {
+		return true, nil
+	}
+
+	if isNotFoundError(err) {
+		return false, nil
+	}
+
+	return false, logAPIError("Error retrieving resource", err)
```
Comment (2ez4szliu) [inline nsxt/resource_nsxt_policy_ipsec_vpn_dpd_profile.go:75]: shall we specify resource type?

File: nsxt/resource_nsxt_policy_ipsec_vpn_dpd_profile.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+func resourceNsxtPolicyIPSecVpnDpdProfileUpdate(d *schema.ResourceData, m interface{}) error {
+	connector := getPolicyConnector(m)
+
+	id := d.Id()
+	if id == "" {
+		return fmt.Errorf("Error obtaining IPSecVpnDpdProfile ID")
+	}
+
+	// Read the rest of the configured parameters
+	description := d.Get("description").(string)
```
Comment (2ez4szliu) [inline nsxt/resource_nsxt_policy_ipsec_vpn_dpd_profile.go:156]: Are these type casting needed? Seems calling Get on schema will return the correct data type
https://github.com/hashicorp/terraform/blob/v0.14.11/helper/schema/schema.go#L65

File: nsxt/resource_nsxt_policy_ipsec_vpn_dpd_profile.go (hunk 1 above)
Comment (annakhm) [reply nsxt/resource_nsxt_policy_ipsec_vpn_dpd_profile.go:75]: terraform would inform what resource is causing the error, so I think we can remove resource type from other places as well..

File: nsxt/resource_nsxt_policy_ipsec_vpn_dpd_profile.go (hunk 2 above)
Comment (annakhm) [reply nsxt/resource_nsxt_policy_ipsec_vpn_dpd_profile.go:161]: `Get` returns `interface{}`, so unless we type cast here compiler would complain..

**Review by ksamoray — APPROVED** [review]

**Review by 2ez4szliu — APPROVED** [review]

**Pull request merged**